### ISSUE TITLE
kube-apiserver liveness probe fails even though it is working

### ISSUE THREAD

**Issue by robertgzr** [kinvolk/kubeadm-nspawn#15]
kubelet logs:
```
kuberuntime_manager.go:458] Container {Name:kubeapiserver (...) 
LivenessProbe:&Probe{Handler:Handler{
Exec:nil,
HTTPGet:&HTTPGetAction{
   Path:/healthz,
   Port:6443,
   Host:127.0.0.1,
   Scheme:HTTPS,
   HTTPHeaders[],
},
TCPSocket:nil,},
InitialDelaySeconds:15,
TimeoutSeconds:15,
PeriodSeconds:10,
SuccessThreshold:1,
FailureThreshold:8,} (...) } is dead, but RestartPolicy says that we should restart it.
```

**Comment by robertgzr** [comment]
This needs more testing. It currently works for me with 
kubelet version `Kubernetes v1.7.0-alpha.1.220+f3d2ea5dfdb893`

**Comment by robertgzr** [comment]
So I found the reason why it was failing:

We set `--anonymous-auth=false` as a apiserver flag in https://github.com/kinvolk/kubeadm-nspawn/blob/robertgzr/spawn-nodes/scripts/bootstrap.sh#L33

Removing this solves it.

**Comment by robertgzr** [comment]
Fixed by #8
